ext/drreg/drreg.c

```diff
@@ -1948,13 +3087,44 @@ drreg_init(drreg_options_t *ops_in)
         ops.error_callback = ops_in->error_callback;
 
     if (prior_slots > 0) {
-        if (!dr_raw_tls_cfree(tls_slot_offs, prior_slots))
+        /* +1 for the pointer to the indirect spill block, see below. */
+        if (!dr_raw_tls_cfree(tls_simd_offs, prior_slots + 1))
             return DRREG_ERROR;
     }
+
     /* 0 spill slots is supported and just fills in tls_seg for us. */
-    if (!dr_raw_tls_calloc(&tls_seg, &tls_slot_offs, ops.num_spill_slots, 0))
+    /* We are allocating an additional slot for the pointer to the indirect
+     * spill block.
+     */
+    if (!dr_raw_tls_calloc(&tls_seg, &tls_simd_offs, ops.num_spill_slots + 1, 0))
         return DRREG_ERROR_OUT_OF_SLOTS;
 
+#ifdef SIMD_SUPPORTED
+    if (prior_simd_slots < ops.num_spill_simd_slots) {
+        /* Refresh init_pt */
+
+        byte *simd_spill_start;
+        byte *simd_spills;
+        tls_alloc_simd_slots(GLOBAL_DCONTEXT, &simd_spill_start, &simd_spills,
```

Comment:
Shouldn't the free below come before the re-alloc?

Reply:
Do you mean store the content on the stack, free and then alloc?

Reply:
Sorry I missed that you're assigning to a local. Forget what I said, it's good as-is. Resolving.